ukis_pysat/raster.py

```diff
@@ -195,13 +195,15 @@ def __update_dataset(self, crs, transform, nodata=None):
             "nodata": nodata,
             "height": self.__arr.shape[-2],
             "width": self.__arr.shape[-1],
-            "count": self.__arr.shape[0],
+            "count": 1 if self.__arr.ndim == 2 else self.__arr.shape[0],
             "crs": crs,
             "transform": transform,
         }
 
         memfile = MemoryFile()
         with memfile.open(**meta) as ds:

```

Comment:
I was just wondering why we don't apply the check [here](https://github.com/dlr-eoc/ukis-pysat/blob/a8bbb98b586bc2429a85dbf0084c2619ee437838/ukis_pysat/raster.py#L55) already where we also check for dimorder, transform and crs? At a first look, this seems more consistent and possibly simpler.

Reply:
We only need it here, when this method is not called there's no need to do it. Your approach is less hidden though.

Reply:
Both options are fine I guess and I would leave it up to you to decide where we should put this. Another benefit of doing the check earlier together with the dimorder check would be that we only do this once and not every time when __update_dataset is called.

Reply:
I agree, this would make it more consistent because we would always return a 3D array.